### PR TITLE
Small optimize code and fix typos

### PR DESCRIPTION
**Pull request by sergeylanzman**
1. Change from deprecated Core to CoreV1 for kube client
2. Fix typo GetAvilableMachineTypes() => GetAvailableMachineTypes() 
3. Fix small typos 
4. Small optimize code

### CODE REVIEW

File: cluster-autoscaler/utils/kubernetes/ready.go [hunk 1]
```diff
@@ -48,23 +48,14 @@ func GetReadinessState(node *apiv1.Node) (isNodeReady bool, lastTransitionTime t
 			if cond.Status == apiv1.ConditionFalse || cond.Status == apiv1.ConditionUnknown {
 				canNodeBeReady = false
 			}
-			if lastTransitionTime.Before(cond.LastTransitionTime.Time) {
-				lastTransitionTime = cond.LastTransitionTime.Time
-			}
 		case apiv1.NodeOutOfDisk:
```
Comment (mwielgus) [inline cluster-autoscaler/utils/kubernetes/ready.go]: In Go empty actions are ... empty actions contrary to some other languages. Please check https://github.com/golang/go/wiki/Switch. So this change is wrong.

Comment (sergeylanzman) [reply cluster-autoscaler/utils/kubernetes/ready.go]: Yes you right, but we have less code. Not need write empty actions.
If you want I can revert this

**Comment by sergeylanzman** [comment]
@mwielgus I reverted switch, and test pass.

**Pull request merged**